src/main/java/com/google/sps/app/servlets/AddUsersToEvent.java

```diff
@@ -0,0 +1,146 @@
+package com.google.sps.app.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.Value;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+import com.google.datastore.v1.client.DatastoreHelper;
+import com.google.gson.Gson;
+import com.google.sps.model.UserList;
+import com.google.sps.util.DataStoreHelper;
+import com.google.sps.util.Validator;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/addUserToEvent")
+public class AddUsersToEvent extends HttpServlet {
+    
+    private static String EVENT_ID = "javascript-input-event-id";
+    private static String USERS = "text-input-user-list";
+
+    /**
+     * Reads an events userlist and adds users that aren't there yet.
+     * @param request. Containing an event ID and
+     * a comma seperated list of usernames too add.
+     * @param response
+     * @return response containing a JSON representation of the new username list
+     * and a list of usernames that failed to add, with an error message.
+     * @throws IOException
+     */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        //get paramaters
+        String id = request.getParameter(EVENT_ID);
+        List<String> usersToAdd = cleanInput(request.getParameter(USERS));
+        
+        //get lists
+        Entity eventEntity = getEventEntity(id);
+        List<String> usernames = getAssociatedEventUsernames(eventEntity);
+        List<String> userIDs = getAssociatedEventUserIDs(eventEntity);
+
+        //add to lists
+        UserList resp = addUsersToExistingUsers(usersToAdd, usernames, userIDs);
+
+        pushUserstoEvent(eventEntity, resp);
```

Comment:
nit: pushUsersToEvent or addUsers?

Reply:
This part updates data store. Could a good name be updatesDatastoreEvent()